pyensign/ensign.py

```diff
@@ -157,11 +183,57 @@ async def archive_topic(self, id):
         raise NotImplementedError
 
     async def destroy_topic(self, id):
-        raise NotImplementedError
+        """
+        Completely destroy a topic including all of its data. This operation is
+        asynchronous so it may take some time for the topic to be destroyed.
+
+        Caution! This operation is irreversible.
+
+        Parameters
+        ----------
+        id : str
+            The ID of the topic to destroy.
+
+        Returns
+        -------
+        bool
+            True if the request was accepted, False otherwise.
+        """
+
+        _, state = await self.client.destroy_topic(id)
+        return state == topic_pb2.TopicTombstone.Status.DELETING
 
     async def topic_names(self):
         raise NotImplementedError
 
+    async def topic_id(self, name):
+        """
+        Get the ID of a topic by name.
+
+        Parameters
+        ----------
+        name: str
+            The name of the topic.
+
+        Returns
+        -------
+        str
+            The ID of the topic or None.
+        """
+
+        # TODO: cache topic IDs to avoid repeated API calls
```

Comment:
:+1:  I know we just did this in the Go SDK but I'm guessing it will take a little extra magic to make it work in Python